design/core_policies.md

```diff
@@ -0,0 +1,31 @@
+# Node.js core Security Policies
+
+The goal of that document is to design a behavior in Node.js core that could be used to disable certain features of
+Node.js.
+
+A simple way to reach that goal is to prevent the loading of certain core modules according to the defined policy.
+
+A policy is enforced by setting up a runtime flag for the Node.js process to disable a set of features:
+
+```sh
+$ node --disable-addons index.js
+```
+
+Also, a per core module blacklist can be set up:
+
+```
+$ node --disable=net,http,https,http2,dns,udp index.js
+```
+
+Question:
+* Do we want to allow policy definition from a file
+* How can we make sure modules like linters, that people start from `.bin` never have access to network
```

Comment:
This should discuss if the scope of permissions is on files within `node_modules/.bin` or if it is in `node_modules/linter/*`. In particular I have concerns about if the permissions are self defined within `node_modules/linter` in a way that allows `linter` to alter the permissions after the first run. I think establishing the authority structure is potentially going to need more care than the feature flags.